api/tests/tooling/chai-custom-helpers/deep-equal-array.js

```diff
@@ -0,0 +1,28 @@
+module.exports = function(chai, _utils) {
+  chai.Assertion.addMethod('deepEqualArray', function(referenceArray) {
+    const assertedArray = this._obj;
+
+    _assertIsArray(chai, assertedArray);
+    _assertIsArray(chai, referenceArray);
+    _assertArraysHaveSameLength(chai, assertedArray, referenceArray);
+
+    for (let i = 0; i < assertedArray.length; ++i) {
+      _assertDeepEqualInstance(chai, assertedArray[i], referenceArray[i]);
+    }
+  });
+};
+
+function _assertIsArray(chai, value) {
+  const instanceClassName = value.constructor.name;
+  new chai.Assertion(instanceClassName).to.equal('Array');
```

Comment:
Ceci ne fonctionne pas ? 
```suggestion
function _assertIsArray(chai, value) {
  new chai.Assertion(value).to.be.instanceOf(Array);
```

Reply:
Ah si ça marche nickel, je change